### ISSUE TITLE
flymake-collection-shellcheck does not work with zsh

### ISSUE THREAD

**Issue by alternateved** [mohkale/flymake-rest#5]
Hello!

I am not sure how exactly troubleshoot this, but it seems that `flymake-collection-shellcheck` does not work (presumably) with other shells. This does not happen with [flymake-shellcheck](https://github.com/federicotdn/flymake-shellcheck) where checker is correctly loaded. 

I noticed it when editing configuration file for `zsh` (.zshrc), where checker wouldn't load unless shebang like `#!/bin/env sh` was added. It wouldn't work with `#!/bin/zsh`. Changing shell for `sh-mode` with `sh-set-shell` from `zsh` to `sh` and reloading `flymake` also makes checker to load. 

Would there be any way to load it properly for zsh?

Best wishes,
alternateved

**Comment by mohkale** [comment]
> I noticed it when editing configuration file for zsh (.zshrc), where checker wouldn't load unless shebang like #!/bin/env sh was added. It wouldn't work with #!/bin/zsh. Changing shell for sh-mode with sh-set-shell from zsh to sh and reloading flymake also makes checker to load.

Yeah, I've run into this issue before as well. It's caused by 2 things:
1. Firstly shellcheck doesn't support zsh :cry:. If you run `shellcheck --help` the --shell option supports sh, bash, dash and ksh. 
2. We just pass the value of `sh-shell` directly to shellcheck as this argument and if you're writing a zshell script then yeah, it'll probably fail.

Personally I don't write zshell specific scripts. Always either sh or bash. The recommended way to fix this IMO would be to:
1. Add a new option `flymake-collection-shellcheck-default-shell` which will be used if we're currently working with an unknown or unsupported shell script dialect.
2. Make the existing shell resolve logic check if sh-shell is in the supported list and if not use `flymake-collection-shellcheck-default-shell` in its place.

PRs are welcome :-).

**Comment by alternateved** [comment]
> 1. Firstly shellcheck doesn't support zsh cry. If you run `shellcheck --help` the --shell option supports sh, bash, dash and ksh.
> 2. We just pass the value of `sh-shell` directly to shellcheck as this argument and if you're writing a zshell script then yeah, it'll probably fail.

Ah, that is why! Now I see, that makes sense, thank you for explaining.

> PRs are welcome :-).

That does not sound hard to implement. I am not proficient with elisp (as much as I would want to), but if I find some time I'll give it a try. Thanks!
